pytorch_lightning/callbacks/lr_monitor.py

```diff
@@ -155,12 +154,30 @@ def on_train_epoch_start(self, trainer, *args, **kwargs):
     def _extract_stats(self, trainer, interval: str) -> Dict[str, float]:
         latest_stat = {}
 
-        names = self._find_names(trainer.lr_schedulers, add_lr_sch_names=False)
-        self._remap_keys(names)
-
-        for name, scheduler in zip(self.lr_sch_names, trainer.lr_schedulers):
-            if scheduler["interval"] == interval or interval == "any":
-                opt = scheduler["scheduler"].optimizer
+        if trainer.lr_schedulers:
+            names = self._find_names(trainer.lr_schedulers, add_lr_sch_names=False)
+            self._remap_keys(names)
+
+            for name, scheduler in zip(self.lr_sch_names, trainer.lr_schedulers):
+                if interval in [scheduler["interval"], "any"]:
+                    opt = scheduler["scheduler"].optimizer
+                    param_groups = opt.param_groups
+                    use_betas = "betas" in opt.defaults
+
+                    for i, pg in enumerate(param_groups):
+                        name_and_suffix = self._add_suffix(name, param_groups, i)
+                        lr = self._extract_lr(pg, name_and_suffix)
+                        latest_stat.update(lr)
+                        momentum = self._extract_momentum(
+                            param_group=pg, name=name_and_suffix.replace(name, f"{name}-momentum"), use_betas=use_betas
+                        )
+                        latest_stat.update(momentum)
```

Comment:
this code is duplicated in the else case below. could we abstract this to a common helper function to reduce the risk of divergence?

Reply:
I have now added helper methods